pir/pir-internal/src/main/java/com/duckduckgo/pir/internal/common/PirActionsRunner.kt

```diff
@@ -0,0 +1,944 @@
+import com.duckduckgo.pir.internal.common.RealPirActionsRunner.Command.GetCaptchaSolution
+import com.duckduckgo.pir.internal.common.RealPirActionsRunner.Command.GetEmail
+import com.duckduckgo.pir.internal.common.RealPirActionsRunner.Command.HandleBroker
+import com.duckduckgo.pir.internal.common.RealPirActionsRunner.Command.Idle
+import com.duckduckgo.pir.internal.common.RealPirActionsRunner.Command.LoadUrl
+import com.duckduckgo.pir.internal.common.RealPirActionsRunner.Command.SendCaptchaSolution
+import com.duckduckgo.pir.internal.scripts.BrokerActionProcessor
+import com.duckduckgo.pir.internal.scripts.BrokerActionProcessor.ActionResultListener
+import com.duckduckgo.pir.internal.scripts.models.BrokerAction
+import com.duckduckgo.pir.internal.scripts.models.BrokerAction.Click
+import com.duckduckgo.pir.internal.scripts.models.BrokerAction.EmailConfirmation
+import com.duckduckgo.pir.internal.scripts.models.BrokerAction.Expectation
+import com.duckduckgo.pir.internal.scripts.models.BrokerAction.GetCaptchaInfo
+import com.duckduckgo.pir.internal.scripts.models.BrokerAction.SolveCaptcha
+import com.duckduckgo.pir.internal.scripts.models.DataSource.EXTRACTED_PROFILE
+import com.duckduckgo.pir.internal.scripts.models.ExtractedProfile
+import com.duckduckgo.pir.internal.scripts.models.ExtractedProfileParams
+import com.duckduckgo.pir.internal.scripts.models.PirErrorReponse
+import com.duckduckgo.pir.internal.scripts.models.PirScriptRequestData
+import com.duckduckgo.pir.internal.scripts.models.PirScriptRequestData.UserProfile
+import com.duckduckgo.pir.internal.scripts.models.PirSuccessResponse
+import com.duckduckgo.pir.internal.scripts.models.PirSuccessResponse.ClickResponse
+import com.duckduckgo.pir.internal.scripts.models.PirSuccessResponse.ExpectationResponse
+import com.duckduckgo.pir.internal.scripts.models.PirSuccessResponse.ExtractedResponse
+import com.duckduckgo.pir.internal.scripts.models.PirSuccessResponse.FillFormResponse
+import com.duckduckgo.pir.internal.scripts.models.PirSuccessResponse.GetCaptchaInfoResponse
+import com.duckduckgo.pir.internal.scripts.models.PirSuccessResponse.GetCaptchaInfoResponse.ResponseData
+import com.duckduckgo.pir.internal.scripts.models.PirSuccessResponse.NavigateResponse
+import com.duckduckgo.pir.internal.scripts.models.PirSuccessResponse.SolveCaptchaResponse
+import com.duckduckgo.pir.internal.scripts.models.ProfileQuery
+import com.duckduckgo.pir.internal.scripts.models.asActionType
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+import logcat.logcat
+
+interface PirActionsRunner {
```

Comment:
Moved this to common package which contains everything shared between scan and optout that is not necessarily script specific